Makefile

```diff
@@ -14,6 +14,19 @@ conformance-test: ## Run the conformance test suite
 proto: ## Rebuild protobuf autogenerated code
 	protoc --go_out=pkg api/sbom.proto
 
+.PHONY: proto-gorm
+proto-gorm: gorm.proto proto ## Compile protobufs with protoc-gen-gorm plugin
+	protoc --gorm_out=pkg api/sbom.proto
+
+PROTOC_GORM_VER := d3024d4fa7c9ce9d62e2f2cc0b42faf679211846
+PROTOC_GORM_URL := https://raw.githubusercontent.com/infobloxopen/protoc-gen-gorm/${PROTOC_GORM_VER}/proto/options/gorm.proto
+
+.SILENT: gorm.proto
+gorm.proto:
```

Comment:
Does gorm provide precompiled libraries that we can use instead of importing and doing the code generation ourselves?

Reply:
They do, located [here](https://github.com/infobloxopen/protoc-gen-gorm/blob/d3024d4fa7c9ce9d62e2f2cc0b42faf679211846/options/gorm.pb.go), but I wasn't able to figure out how to access the needed message options (e.g. `option (gorm.opts) = ...`) without importing the proto file.

The proto file gets installed along with the plugin, but I couldn't come up with an intuitive way to find the file path. e.g. with a `PROTOC_GORM_VER` of `d3024d4fa7c9ce9d62e2f2cc0b42faf679211846`, on my system the file ends up located at `~/go/pkg/mod/github.com/infobloxopen/protoc-gen-gorm@v1.1.3-0.20231122062459-d3024d4fa7c9` after the `go install`.

Reply:
@fkautz Do you know of a way to use the protoc-gen-gorm options without importing the proto? How do you want me to resolve this conversation?

Reply:
@veramine @jhoward-lm Here's my understanding:

Gorm lacks a straightforward method for importing the gorm.proto. They suggest incorporating it into a build process, as you've done. Previously, with older Go versions, the package could be fetched into GOPATH/pkg/src, allowing you to map it during protoc builds.

My suggestion is to use a .tmp directory for build files that should be ignored. Additionally, I prefer having the entire repository cloned, as it contains many other protos we may need to reference.

I've attached an example patch (although I lack permission to suggest it to this branch). What are your thoughts?

[gorm_import.patch.txt](https://github.com/bom-squad/protobom/files/14330359/gorm_import.patch.txt)

Reply:
Would something like this do the same thing as cloning with `git`? I’m not exactly a Go expert, maybe `GOBIN` instead of `GOPATH`?

```makefile
ROOT_DIR ?= $(abspath $(dir $(lastword $(MAKEFILE_LIST))))

GOPATH=$(ROOT_DIR)/.tmp go install protoc-gen-gorm@$(PROTOC_GORM_VER)
```
AFAIK, the `protoc-gen-gorm` project really only has 2 `.proto` files meant to be imported by other protobufs ([link](https://github.com/infobloxopen/protoc-gen-gorm/tree/main/proto)). Since this PR adds it as a project dependency, I would think that all of the files in that repo should be locally available to devs once `go install` is run.

Here’s the patch text in a ` ```diff` code block:

<details>

<summary>Expand</summary>

```diff
diff --git a/.gitignore b/.gitignore
index 20c3236..ab39967 100644
--- a/.gitignore
+++ b/.gitignore
@@ -20,3 +20,4 @@
 # Go workspace file
 go.work
 .vscode
+*.tmp
diff --git a/Makefile b/Makefile
index f9d5ac0..2039df2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,13 @@
+
+TEMPDIR = ./.tmp
+
+ifndef TEMPDIR
+	$(error TEMPDIR is not set)
+endif
+
+$(TEMPDIR):
+	mkdir -p $(TEMPDIR)
+
 .PHONY: help
 help:
 	@grep -Eh '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "$(BOLD)$(CYAN)%-25s$(RESET)%s\n", $$1, $$2}'
@@ -14,18 +24,21 @@ conformance-test: ## Run the conformance test suite
 proto: ## Rebuild protobuf autogenerated code
 	protoc --go_out=pkg api/sbom.proto
 
-.PHONY: proto-gorm
-proto-gorm: gorm.proto proto ## Compile protobufs with protoc-gen-gorm plugin
-	protoc --gorm_out=pkg api/sbom.proto
+.PHONY: gorm proto
+proto-gorm: gorm ## Compile protobufs with protoc-gen-gorm plugin
+	protoc --go_out=pkg --gorm_out=. api/sbom.proto -I. -I$(TEMPDIR)
 
 PROTOC_GORM_VER := d3024d4fa7c9ce9d62e2f2cc0b42faf679211846
 PROTOC_GORM_URL := https://raw.githubusercontent.com/infobloxopen/protoc-gen-gorm/${PROTOC_GORM_VER}/proto/options/gorm.proto
 
-.SILENT: gorm.proto
-gorm.proto:
+.SILENT: gorm
+gorm:
 	echo Downloading protoc-gen-gorm...
+	rm -rf $(TEMPDIR)/protoc-gen-gorm | true
+	git clone https://github.com/infobloxopen/protoc-gen-gorm.git $(TEMPDIR)/protoc-gen-gorm --depth 1 --branch main --single-branch
 	go install github.com/infobloxopen/protoc-gen-gorm@$(PROTOC_GORM_VER)
-	curl --fail --silent --show-error --location --remote-name --url $(PROTOC_GORM_URL)
+	
+# curl --fail --silent --show-error --location --remote-name --url $(PROTOC_GORM_URL)
 
 .PHONY: fakes
 fakes: ## Rebuild the fake implementations
diff --git a/api/sbom.proto b/api/sbom.proto
index 24b3b07..896c5c9 100644
--- a/api/sbom.proto
+++ b/api/sbom.proto
@@ -1,7 +1,8 @@
 syntax = "proto3";
 
-import "gorm.proto";
+
 import "google/protobuf/timestamp.proto";
+import "protoc-gen-gorm/proto/options/gorm.proto";
 
 option go_package = "sbom/";
 package bomsquad.protobom;
```

</details>

I would probably do `$(RM) -r` and have it in a separate clean target but it’s just a style thing

Reply:
If it functions correctly, then yes, I attempted it with the latest version of Go but couldn't get the sources to be fetched by the Go binary, so I resorted to pulling them with Git.

As far as I understand, GOPATH is a somewhat backward-compatible feature, whereas GOBIN is typically just `$GOPATH/bin`. So, in reality, I don't think it will make much of a difference.

Regarding the Git clone, I suppose we can only fetch the protocol, but I thought it would be clearer how to add other protocol source repositories if needed when cloning the entire repository.

Yes, indeed, we can remove the `rm` command and instead move it to a different make target. I just outlined it quickly.